### PR TITLE
Do not throw errors for endAsync being called multiple times

### LINKED ISSUE #4: Determine whether we need to track when `endAsync` is called more than once.
We currently only error if `endAsync` is called without calling a `beginAsync` in the `TestWaiter` class. It may be the case that we find we need more fine grained error messaging for two scenarios:

1. `endAsync` is called before `beginAsync` is called, thus not registering the waiter item (implemented)
2. `endAsync` is called more than once for the same waiter item (not implemented).

### PR DESCRIPTION
**Pull request by seanjohnson08**
Fixes #4

### CODE REVIEW

File: addon/test-waiter.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ export default class TestWaiter<T = Token> implements ITestWaiter<T> {
   private isRegistered = false;
 
   items = new Map<T, ITestWaiterDebugInfo>();
+  endedItems = new Map<T, boolean>();
```
Comment (rwjblue) [inline addon/test-waiter.ts]: Can we avoid needing the extra map here, by weakly associating the `TestWaiter` with the token?

Comment (scalvert) [reply addon/test-waiter.ts]: To be clear, Rob is suggesting we do

```ts
let trackedTokens = new WeakMap<Token, TestWaiter>();
```

to track the association of the token with a particular test waiter instance, _and_ have the ability to detect if a token is attempting to be used to invoke `endAsync` more than once.

**Comment by rwjblue** [comment]
@scalvert / @seanjohnson08 - Where are we at with this one?

**Comment by scalvert** [comment]
@seanjohnson08?

**Comment by seanjohnson08** [comment]
I'm closing this - I don't believe it's an issue for our use case any longer.

**Pull request closed without merging**